epigraphhub/data/worldbank.py

```diff
@@ -0,0 +1,257 @@
+
+    else:
+
+        keyword = keyword.replace(" ", "").lower()
+
+        for source in wb.source.list():
+
+            if keyword in source["name"].replace(" ", "").lower():
+
+                results.append(source)
+
+    return pd.DataFrame.from_records(results)
+
+
+def search_in_indicators(keyword, db=2):
+
+    """
+    Returns a dataframe with the indicators matched by partial name.
+
+    :params keyword: string. keyword to search in the indicators name.
+    :params db:int. Number associated with the database whose you want to get the list
+                    of indicators. You can discover this number in the function 'search_in
+                    _database'. By default the indicators are search over the World Development
+                    Indicators database (db = 2).
+
+    :returns: DataFrame
+
+    """
+
+    results = []
+
+    for indicator in wb.series.list(q=keyword, db=db):
+        results.append(indicator)
+
+    return pd.DataFrame.from_records(results)
+
+
+def get_worldbank_data(ind, country, db=2, time="all", columns=None):
+
+    """
+    This function get a list of indicators according to some country from the world data bank
+    and return this series in a dataframe.
+
+    :params ind: List of strings. List with the indicators whose data you want to get.
+
+    :params country: List of strings|string. List with the ISO-CODE for the countries whose
+                     data you want to get.
+
+    :params time: Interval of  years. If filled `time = 'all'`, the function will
+                  return all the data available. You can also specify a range of years.
+                  For example, if you want to get the data for the period between the
+                  years 2010 and 2020, you can fill this parameter with
+                  `time = range(2010,2021)`.
+
+    :params columns: List of strings. List with names to rename the columns instead of use the
+                    names in the `ind` list.
+
+    Important: The lists `ind` and `columns` must have the same lenght.
+
+    :returns: DataFrame
+
+    """
+
+    if columns != None:
+
+        if len(columns) == len(ind):
+
+            rename_columns = {}
+
+            for i in np.arange(0, len(ind)):
+                rename_columns[ind[i]] = columns[i]
+
+    else:
+
+        rename_columns = {}
+
+    if len(ind) == 1:
+
+        df = wb.data.DataFrame(
+            series=ind, economy=country, db=db, index=["time"], labels=True, time=time
+        )
+
+        df = df.reset_index()
+
+        del df["time"]
+
+        df["Time"] = df["Time"].astype("str")
+
+    else:
+        df = wb.data.DataFrame(
+            series=ind,
+            economy=country,
+            db=db,
+            index=["time", "economy"],
+            labels=True,
+            time=time,
+        )
+
+    if len(rename_columns) != 0:
+
+        df = df.rename(columns=rename_columns)
+
+    new_dates = []
+    periodicity = []
+
+    for i in df["Time"]:
```

Comment:
instead of this for loop, the efficient way would be to use `pd.date_range` with the correct periodicity.

Reply:
The problem with this solution is that although most data are annual, as explained in the comment above, we have cases where there is monthly and quarterly data.

Even in the case that the data are necessarily annual, this data is not available with a constant frequency. I found indicators that return data for sparse years like, 2000, 2005, 2010,2011,2012,2013...

Reply:
In this case you can improve your parser function and perhaps do something like this:
```python
df['format'] = 1
df.loc[df.Time.str.contains('M'), 'format'] = 2
df['new_date'] = pd.to_datetime(df.Time)

# Convert to datetime with two different format settings
df.loc[df.format == 1, 'new_date'] = pd.to_datetime(df.loc[df.format == 1, 'Time'])
df.loc[df.format == 2, 'new_date'] = pd.to_datetime(df.loc[df.format == 2, 'Time'], format = '%YM%m')
```